src/lib/parser/core/PokemonSaveParser.ts

```diff
@@ -624,6 +627,133 @@ export class PokemonSaveParser {
   getWebSocketClient (): MgbaWebSocketClient | null {
     return this.webSocketClient
   }
+
+  /**
+   * Start watching for Pokemon party changes in memory mode
+   * This will automatically set up memory watching for the party data regions
+   * and call the provided callback when changes are detected
+   */
+  async watch (options: {
+    onPartyChange?: (partyPokemon: PokemonBase[]) => void
+    onError?: (error: Error) => void
+  } = {}): Promise<void> {
+    if (!this.isMemoryMode || !this.webSocketClient) {
+      throw new Error('Watch mode only available in memory mode (WebSocket connection)')
+    }
+
+    if (this.isWatching) {
+      throw new Error('Already watching memory. Call stopWatching() first.')
+    }
+
+    // Add callback to listeners if provided
+    if (options.onPartyChange) {
+      this.watchListeners.push(options.onPartyChange)
+    }
+
+    // Set up memory watching for party data regions
+    if (!this.config?.memoryAddresses?.preloadRegions) {
+      throw new Error('No memory addresses configured for watching')
+    }
+
+    const partyRegions = this.config.memoryAddresses.preloadRegions
+
+    try {
+      // Start watching memory regions directly
+      await this.webSocketClient.startWatching(partyRegions)
+
+      // Set up memory change listener
+      const memoryChangeListener = async (address: number, size: number, data: Uint8Array) => {
+        try {
+          // Use the provided data directly instead of making additional calls
+          await this.handlePartyDataChangeWithData(address, size, data)
+        } catch (error) {
+          if (options.onError) {
+            options.onError(error instanceof Error ? error : new Error(String(error)))
+          }
+        }
+      }
+
+      this.webSocketClient.addMemoryChangeListener(memoryChangeListener)
+      this.isWatching = true
+
+      // Load initial data to establish baseline
+      await this.handlePartyDataChange()
+    } catch (error) {
+      if (options.onError) {
+        options.onError(error instanceof Error ? error : new Error(String(error)))
+      } else {
+        throw error
+      }
+    }
+  }
+
+  /**
+   * Handle party data changes with provided memory data and notify listeners
+   */
+  private async handlePartyDataChangeWithData (address: number, size: number, data: Uint8Array): Promise<void> {
+    if (!this.webSocketClient) return
+
+    try {
+      const partyPokemon = await this.parsePartyPokemon()
+
+      // Notify all listeners (no need for change detection as server only sends when changed)
+      for (const listener of this.watchListeners) {
+        try {
+          listener(partyPokemon)
+        } catch (error) {
+          console.error('Watch listener error:', error)
+        }
+      }
+    } catch (error) {
+      console.error('Error handling party data change:', error)
+    }
+  }
```

Comment:
we do not want to parse pokemon here. rather we wan to patch the view of our party pokemon. also remove dupe code

Reply:
please check the pokemon data class on how to do this

Reply:
can we architect it in such a way, that we do not need to cache the Pokémon party?